### PR TITLE
 Allow booting from ethereum snapshot

### PR DESCRIPTION
**Pull request by philipcmonk**
Pairs with urbit/urbit#1046.  Usage described there.

Definitely take a peek at my tests, I'm not used to writing them, and I'm trying to get in the habit.

### CODE REVIEW

File: sys/zuse.hoon [hunk 1]
```diff
@@ -8135,6 +8161,54 @@
...
+    ++  czar
+      |=  snap=snapshot:jael
+      ^-  (map ship [=life =pass])
+      %-  malt
+      %+  turn  (gulf 0 255)
+      |=  gal=@
+      ^-  [ship [life pass]]
+      :-  gal
+      ~|  czar-gal=gal
+      [life pass]:(need net:(~(got by hul.eth.snap) gal))
```
[the remark this answers is not in the document]
Comment (joemfb) [reply sys/zuse.hoon]: I expect we'll want to make individual galaxies optional here. But that can easily be done later.

File: sys/zuse.hoon [hunk 2]
```diff
@@ -8135,6 +8161,54 @@
...
+      ^-  (list ^turf)
+      %+  murn
+        ^-  (list (pair))
+        %+  murn
+          ^-  (list @t)
+          ~[pri sec ter]:dns.eth.snap
+        |=  dom=@t
+        ^-  (unit (pair))
+        (rush dom thos:de-purl:html)
+      |=([* a=*] ((soft ^turf) a))
```
Comment (joemfb) [inline sys/zuse.hoon]: For compatibility with `+turf:take:dawn`, this should filter out duplicates. Also not pressing.

**Comment by philipcmonk** [comment]
Going forward, is `ames-eth-keys` deprecated and I should PR against `release-candidate`?

**Comment by joemfb** [comment]
Yeah, that sounds good. We had a growing collection of backwards-incompatible changes for a while, but, barring that, my preference is to integrate into release-candidate as quickly as possible.

**Comment by philipcmonk** [comment]
What counts as backwards-incompatible?  E.g. if I change the type of a vane's state, so a solid is required, is that backwards-incompatible?

**Comment by joemfb** [comment]
That was a poor choice of words -- I just meant that ames-eth-keys in both Arvo and vere was a very disruptive set of changes that could be integrated without actually (mostly) working. No overarching standard implied ... ;)

**Pull request merged**